File: dm-backend/Models/FaultyDeviceModel.cs
using System;
using System.Runtime.Serialization;

namespace dm_backend.Models
{
    [DataContract]
    public class FaultyDeviceModel
    {
        

        [DataMember]
        public int complaintId { get; set; }
        [DataMember]
        public int deviceId { get; set; }
        [DataMember]
        public string deviceType { get; set; }
        [DataMember]
        public string deviceBrand { get; set; }
        [DataMember(Name="device")]
        public string device { get; set; }
        [DataMember]
        public string deviceModel { get; set; }
        [DataMember(Name="serialNumber")]
        public string serialNumber { get; set; }
        [DataMember]

        public string salutation { get; set; }

        public PartialUserModel userName { get; set; }
        [DataMember(Name="name")]
        public string name{get; set;}
        [DataMember(Name="complaintDate")]
        public DateTime complaintDate { get; set; }
        [DataMember]
        public string Comments { get; set; }
        [DataMember]
        public string status { get; set; }
        [DataMember]
        public string image { get; set; }
   

    }
   
}

File: dm-frontend/faultyDevice/FaultyDeviceModel.ts
export class FaultyDeviceModel
{
    public complaintId : number
    public userId : number  
    public name : string
    public deviceId : number 
    public device : string 
    public serialNumber : string 
    public complaitDate  : string 
    public Comments : string
    public image:string


    constructor(data : any )
    {
        this.complaintId = data.complaintId;
        this.userId = data.userId;
        this.deviceId = data.deviceId;
        this.serialNumber= data.serialNumber;
        this.name = data.name;
        this.device =  data.device;
        this.Comments = data.Comments;
        this.complaitDate = data.complaintDate;
        this.image=data.image;
    }
}